pkg/csi_driver/multishare_controller.go

```diff
@@ -228,30 +216,121 @@ func (m *MultishareController) startAndWaitForInstanceDeleteOrShrink(ctx context
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	klog.Infof("ControllerExpandVolume called for multishare with request %+v", req)
+	if acquired := m.volumeLocks.TryAcquire(volumeId); !acquired {
+		return nil, status.Errorf(codes.Aborted, util.VolumeOperationAlreadyExistsFmt, volumeId)
+	}
+	defer m.volumeLocks.Release(volumeId)
+
+	share, err := m.cloud.File.GetShare(ctx, &file.Share{
+		Parent: &file.MultishareInstance{
+			Project:  project,
+			Location: location,
+			Name:     instanceName,
+		},
+		Name: shareName,
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	if share == nil || file.IsNotFoundErr(err) {
+		return nil, status.Errorf(codes.Internal, "Couldn't find share with name %q", volumeId)
+	}
+
+	if share.CapacityBytes >= reqBytes {
+		klog.Infof("Controller expand volume succeeded for volume %v, existing size(bytes): %v", volumeId, share.CapacityBytes)
+		return &csi.ControllerExpandVolumeResponse{
+			CapacityBytes:         share.CapacityBytes,
+			NodeExpansionRequired: false,
+		}, nil
+	}
+
+	workflow, err := m.opsManager.expandInstanceAndStartWorkflow(ctx, share, reqBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	err = m.waitOnWorkflow(ctx, workflow)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "wait on operation %q failed with error: %v", workflow.opName, err)
+	}
+	klog.Infof("Wait for operation %s (type %s) completed", workflow.opName, workflow.opType.String())
+
+	switch workflow.opType {
+	case util.InstanceUpdate:
+		workflow, err = m.opsManager.startShareExpandWorkflowSafe(ctx, share, reqBytes)
+		if err != nil {
+			return nil, err
+		}
+	case util.ShareUpdate:
+		share, err = m.cloud.File.GetShare(ctx, share)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, err.Error())
+		}
+		return &csi.ControllerExpandVolumeResponse{
+			CapacityBytes:         share.CapacityBytes,
+			NodeExpansionRequired: false,
+		}, nil
+	default:
+		return nil, status.Errorf(codes.Internal, "Controller Expand Volume failed, unknown workflow %v detected", workflow.opType)
+	}
+
+	err = m.waitOnWorkflow(ctx, workflow)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "wait on expansion %q failed with error: %v", workflow.opName, err)
+	}
+
+	share, err = m.cloud.File.GetShare(ctx, share)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	return &csi.ControllerExpandVolumeResponse{
+		CapacityBytes:         share.CapacityBytes,
+		NodeExpansionRequired: false,
+	}, nil
+}
+
+func (m *MultishareController) waitOnWorkflow(ctx context.Context, workflow *Workflow) (err error) {
+	timeout, pollInterval, err := util.GetMultishareOpsTimeoutConfig(workflow.opType)
+	if err != nil {
+		return
+	}
+	err = m.cloud.File.WaitForOpWithOpts(ctx, workflow.opName, file.PollOpts{Timeout: timeout, Interval: pollInterval})
```

Comment:
we can log the error here with workflow op name and type

Reply:
Isn't the error already logged by whoever calling this func

Reply:
let's revisit the logging in general as a different PR